helm-chart/renku-notebooks/templates/statefulset.yaml

```diff
@@ -194,8 +194,17 @@ spec:
       nodeSelector:
 {{ toYaml . | indent 8 }}
     {{- end }}
-    {{- with .Values.affinity }}
       affinity:
+        podAntiAffinity:
```

Comment:
This section should look like this:

```yaml
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
            - weight: 100
              podAffinityTerm:
                labelSelector:
                  matchExpressions:
                    - key: app
                      operator: In
                      values:
                        - {{ template "notebooks.name" . }}
                topologyKey: "kubernetes.io/hostname"
```

The `weight` field is mandatory. And you are missing the `podAffinityTerm` level.

Reply:
p.s. can you also please tag with `/deploy #persist` after you make the changes?

Reply:
Right I mixed up `preferredDuringSchedulingIgnoredDuringExecution` where it's required and `requiredDuringSchedulingIgnoredDuringExecution` where it's not.

Though if we should have a strong preference for it or require it is an open question